Tasks/Xcode/xcode.ts

```diff
@@ -231,44 +231,61 @@ async function run() {
             xcb.pipeExecOutputToTool(xcPrettyTool);
         }
 
-        //--- Xcode Build ---
-        await xcb.exec();
+        const publishResults: boolean = tl.getBoolInput('publishJUnitResults', false);
+        let xcodeTestErrors;
+        if (publishResults && useXcpretty && actions.toString().toLowerCase().indexOf('test') >= 0) {
```

Comment:
What are the possible options for `actions`?  What is the possibility that there will be a name collision on this?

(Example: 'test' and also 'contested' (just an example), which would be caught by this test.)

Reply:
:) possible. The actions I know of are build, archive, test etc but it is very much possible that a new one will be added. I am moving this to the postexecution job and removing the action.

Reply:
OK.  Another alternative could be to preserve the string array and loop over them.